src/plugins/csp_handler/server/csp_handlers.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ConfigurationClient } from '../../application_config/server';
+import {
+  CoreSetup,
+  IScopedClusterClient,
+  Logger,
+  OnPreResponseHandler,
+  OnPreResponseInfo,
+  OnPreResponseToolkit,
+  OpenSearchDashboardsRequest,
+} from '../../../core/server';
+
+const CSP_RULES_CONFIG_KEY = 'csp.rules';
+
+export function createCspRulesPreResponseHandler(
+  core: CoreSetup,
+  cspHeader: string,
+  getConfigurationClient: (inputOpenSearchClient: IScopedClusterClient) => ConfigurationClient,
+  logger: Logger
+): OnPreResponseHandler {
+  return async (
+    request: OpenSearchDashboardsRequest,
+    response: OnPreResponseInfo,
+    toolkit: OnPreResponseToolkit
+  ) => {
+    try {
+      const shouldCheckDest = ['document', 'frame', 'iframe', 'embed', 'object'];
```

Comment:
where are these element comes from, add a link if any official doc

Reply:
Good callout! Just added.

Reply:
When adding the reference link to code, it triggered a link check failure https://github.com/opensearch-project/OpenSearch-Dashboards/actions/runs/8177983078/job/22360904964?pr=5641

Thus let me directly share the reference link here
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-ancestors